tools/metrics/query.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+)
+
+type queryClient struct {
+	url   string
+	token string
+	hc    *http.Client
+}
+
+func newQueryClient(url, token string, client *http.Client) *queryClient {
+	return &queryClient{url: url, token: token, hc: client}
+}
+
+func newDefaultQueryClient() *queryClient {
+	return newQueryClient(ghAPIDefaultURL, os.Getenv("GITHUB_TOKEN"), &http.Client{})
+}
+
+func (qc *queryClient) query(endpoint string, args map[string]string) (string, error) {
+	qURL, err := buildQueryURL(qc.url, endpoint, args)
+	if err != nil {
+		return "", err
+	}
+	req, err := http.NewRequest(http.MethodGet, qURL, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed prepare GET request for %q: %w", qURL, err)
+	}
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", qc.token))
+	req.Header.Add("X-GitHub-Api-Version", "2022-11-28")
+	rsp, err := qc.hc.Do(req)
```

Comment:
what happens if the request fails?

Reply:
(obviously the answer is that it errors out, my question is more about the bigger picture: will CI retry? will CI fail? should we retry here?)

Reply:
I think for now we can just let fail. This will be run manually, at least initially.

Also I am about to revamp the change completely using a different library.